### PR TITLE
fix/labels

### PR DESCRIPTION
**Pull request by sywhb**
- **do not create label if label name case-insensitively matches existing label name**
- **allow delete internal labels**
- **deduplicate labels when there are case-insensitive identical names**
- **show label_attached instead of label_created in rules**

### CODE REVIEW

File: packages/api/src/services/labels.ts [hunk 1]
```diff
@@ -41,21 +41,18 @@ export const findOrCreateLabels = async (
...
-        labels.map((l) => l.name),
-        userId
-      ),
+    async (tx) => {
+      const repo = tx.withRepository(labelRepository)
+      // create labels if not exist
+      await repo.createLabels(labels, userId)
+
+      // find labels by names
+      return repo.findBy({
```
Comment (jacksonh) [inline packages/api/src/services/labels.ts:51]: is this part case insensitive?

File: packages/api/test/resolvers/labels.test.ts [hunk 2]
```diff
@@ -160,17 +150,30 @@ describe('Labels API', () => {
 
       before(async () => {
         existingLabel = await createLabel('label3', '#ffffff', user.id)
```
Comment (jacksonh) [inline packages/api/test/resolvers/labels.test.ts:152]: Maybe we should create this one as `LABEL3` to test the case sensitivity

File: packages/api/src/services/labels.ts (hunk 1 above)
Comment (sywhb) [reply packages/api/src/services/labels.ts:51]: No, this part is case sensitive so we ensure only one RSS label would be matched if there is another `rss` label created by users

File: packages/api/test/resolvers/labels.test.ts [hunk 3]
```diff
@@ -160,17 +150,30 @@ describe('Labels API', () => {
...
+        }).expect(200)
+
+        expect(res.body.data.createLabel.errorCodes).to.eql([
+          'LABEL_ALREADY_EXISTS',
+        ])
+      })
 
-        expect(res.body.data.createLabel.errorCodes).to.eql(['BAD_REQUEST'])
+      it('returns error code LABEL_ALREADY_EXISTS if case-insensitive label name exists', async () => {
+        const name = existingLabel.name.toUpperCase()
```
Comment (sywhb) [inline packages/api/test/resolvers/labels.test.ts:170]: > Maybe we should create this one as `LABEL3` to test the case sensitivity

It's here 😆

**Review by sywhb — COMMENTED** [review]
Comments

**Review by jacksonh — APPROVED** [review]

**Pull request merged**